src/components/PatronBlock/PatronBlockModal.js

```diff
@@ -8,28 +9,47 @@ import {
   Row
 } from '@folio/stripes/components';
 
-const PatronBlockModal = ({ open, onClose, patronBlocks, viewUserPath }) => (
-  <Modal
-    open={open}
-    onClose={onClose}
-    label={<b><FormattedMessage id="ui-checkout.blockModal" /></b>}
-    dismissible
-  >
-    <Row>
-      <Col xs><FormattedMessage id="ui-checkout.blockedLabel" /></Col>
-    </Row>
-    <Row>
-      <Col xs><b>{patronBlocks.desc || ''}</b></Col>
-    </Row>
-    <br />
-    <Row end="xs">
-      <Col xs>
-        <Button onClick={onClose}><FormattedMessage id="ui-checkout.close" /></Button>
-        <Button style={{ 'marginLeft': '15px' }} buttonStyle="primary" onClick={viewUserPath}><FormattedMessage id="ui-checkout.detailsButton" /></Button>
-      </Col>
-    </Row>
-  </Modal>
-);
+const PatronBlockModal = ({ open, onClose, patronBlocks, viewUserPath }) => {
+  const blocks = take(orderBy(patronBlocks, ['metadata.updatedDate'], ['desc']), 3);
+  const renderBlocks = blocks.map(block => {
+    return (
+      <Row>
+        <Col xs>
+          <b>{block.desc || ''}</b>
+        </Col>
+      </Row>
+    );
+  });
+
+  return (
+    <Modal
+      open={open}
+      onClose={onClose}
+      label={<b><FormattedMessage id="ui-checkout.blockModal" /></b>}
+      dismissible
+    >
+      <Row>
+        <Col xs>
+          <FormattedMessage id="ui-checkout.blockedLabel" />
+:
+        </Col>
+      </Row>
+      {renderBlocks}
+      <br />
+      <Row>
+        <Col xs={8}>{(patronBlocks.length > 3) && <FormattedMessage id="ui-checkout.additionalReasons" />}</Col>
+        <Col xs={4}>
```

Comment:
A cleaner option here is a [ModalFooter](/folio-org/stripes-components/tree/master/lib/ModalFooter).